torch/_subclasses/functional_tensor.py

```diff
@@ -537,6 +559,29 @@ def unwrap(x):
                         torch.Tensor, wrap, outs_unwrapped
                     )
                 else:
+                    arg_mutation_counters = []
+
+                    if self.export:
+                        # Check args before the operation
+                        for arg in args:
+                            if isinstance(
+                                arg, FunctionalTensor
+                            ) and torch._is_functional_tensor(arg.elem):
+                                arg_storage = StorageWeakRef(arg.untyped_storage())
+                                arg_group_id = self._partial_frozen_storage.get(
+                                    arg_storage, None
+                                )
+                                if (
+                                    arg_group_id in self._mutated_partial_frozen_storage
+                                ):  # type: ignore[attr-defined]
+                                    raise RuntimeError(
+                                        "Cannot do computation on mutated frozen tensor"
+                                    )
+                                arg_mutation_counter = torch._mutation_counter(arg.elem)  # type: ignore[attr-defined]
+                                arg_mutation_counters.append(arg_mutation_counter)
+                            else:
+                                arg_mutation_counters.append(0)
```

Comment:
nit: just make this a dict from `arg_index -> mutation_counter`  instead of putting in zeros for non-tensor args